agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/indexer/GeneIndexerRepository.java

```diff
@@ -157,263 +177,295 @@ public void run() {
 	}
 
 	private class GetSynonymsThread implements Runnable {
+		File cacheFile = new File(filePrefix + "_" + getClass().getSimpleName() + ".data");
+
 		@Override
 		public void run() {
 			log.info("Building gene -> synonyms map");
 			String query = "MATCH (species:Species)-[:FROM_SPECIES]-(gene:Gene)-[:ALSO_KNOWN_AS]-(s:Synonym) ";
 			query += " RETURN gene.primaryKey as id, s.name as value ";
-
-			cache.setSynonyms(getMapSetForQuery(query));
+			cache.setSynonyms(getCacheResults(cacheFile, query));
 			log.info("Finished Building gene -> synonyms map");
 		}
 	}
 
 	private class GetCrossReferencesThread implements Runnable {
+		File nameCacheFile = new File(filePrefix + "_" + getClass().getSimpleName() + "_names.data");
+		File idsCacheFile = new File(filePrefix + "_" + getClass().getSimpleName() + "_ids.data");
+
 		@Override
 		public void run() {
+
 			log.info("Building gene -> cross references map");
-			String query = "MATCH (species:Species)-[:FROM_SPECIES]-(gene:Gene)-[:CROSS_REFERENCE]-(cr:CrossReference) ";
-			query += " RETURN gene.primaryKey as id, cr.name as value";
+			try {
 
-			Map<String, Set<String>> names = getMapSetForQuery(query);
+				String query = "MATCH (species:Species)-[:FROM_SPECIES]-(gene:Gene)-[:CROSS_REFERENCE]-(cr:CrossReference) ";
```

Comment:
Can you use text blocks that makes it more readable and usable. You can copy / paste the whole query in one step.
"""
MATCH (species:Species)-[:FROM_SPECIES]-(gene:Gene)-[:CROSS_REFERENCE]-(cr:CrossReference)
RETURN gene.primaryKey as id, c
"""